### PR TITLE
UnitTests: Add test TestDeleteStorageClasses

### PR DESCRIPTION
**Pull request by iamniting**
Add test for deleteStorageClasses in uninstall_reconciler_test.go

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

cc: @raghavendra-talur

### CODE REVIEW

**Review by jarrpa — CHANGES_REQUESTED** [review]
Looks good, the tests make sense. Please implement the test function using case tables.

**Comment by iamniting** [comment]
> Looks good, the tests make sense. Please implement the test function using case tables.

Hi, Jose thanks for looking into it. I did not get what you are saying. Can you pls elaborate?

**Comment by iamniting** [comment]
> > Looks good, the tests make sense. Please implement the test function using case tables.
> 
> Hi, Jose thanks for looking into it. I did not get what you are saying. Can you pls elaborate?

I guess I understood it and changed it.

**Review by raghavendra-talur — APPROVED** [review]

File: pkg/controller/storagecluster/uninstall_reconciler_test.go [hunk 1]
```diff
@@ -102,3 +104,30 @@ func assertCephClusterCleanupPolicy(
 	assert.Equal(t, CleanupPolicyConfirmation, cephCluster.Spec.CleanupPolicy.Confirmation)
 	assert.Equal(t, AllowUninstallWithVolumes, cephCluster.Spec.CleanupPolicy.AllowUninstallWithVolumes)
 }
+
+func TestDeleteStorageClasses(t *testing.T) {
```
Comment (umangachapagain) [inline pkg/controller/storagecluster/uninstall_reconciler_test.go:108]: Use Table Driven approach and simplify the code a little bit.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

**Comment by jarrpa** [comment]
> > > Looks good, the tests make sense. Please implement the test function using case tables.
> > 
> > 
> > Hi, Jose thanks for looking into it. I did not get what you are saying. Can you pls elaborate?
> 
> I guess I understood it and changed it.

Sadly, no. :) As Umanga also says, follow the pattern of using a struct that is a list of test cases, as is done here: https://github.com/openshift/ocs-operator/pull/797/files

**Comment by iamniting** [comment]
> > > > Looks good, the tests make sense. Please implement the test function using case tables.
> > > 
> > > 
> > > Hi, Jose thanks for looking into it. I did not get what you are saying. Can you pls elaborate?
> > 
> > 
> > I guess I understood it and changed it.
> 
> Sadly, no. :) As Umanga also says, follow the pattern of using a struct that is a list of test cases, as is done here: https://github.com/openshift/ocs-operator/pull/797/files

It should be resolved now

File: pkg/controller/storagecluster/uninstall_reconciler_test.go (hunk 1 above)
Comment (iamniting) [reply pkg/controller/storagecluster/uninstall_reconciler_test.go:108]: done

File: pkg/controller/storagecluster/uninstall_reconciler_test.go [hunk 2]
```diff
@@ -102,3 +104,53 @@ func assertCephClusterCleanupPolicy(
...
+
+	testList := []struct {
+		isStorageClassExistBeforedelete bool
+		isStorageClassExistAfterdelete  bool
+	}{
+		// case 1: verify storage classes are present and delete them
+		{true, false},
+		// case 2: verify storage classes does not exist and delete should not get error out
+		{false, false},
+	}
```
Comment (jarrpa) [inline pkg/controller/storagecluster/uninstall_reconciler_test.go:118]: This is the right general idea, but it is too condensed to be immediately readable. I suggest something like this:

```suggestion
	cases := []struct {
		label string
		storageClassExists bool
	}{
		{
			label: "case 1", // verify storage classes are present and delete them
			storageClassExists: true,
		},
		{
			label: "case 2", // verify storage classes does not exist and delete should not get error out
			storageClassExists: false,
		},
	}
```

**Review by jarrpa — CHANGES_REQUESTED** [review]
An improvement, but I think the test itself is a little too convoluted. Rename `isStorageClassExistBeforedelete` to `storageClassExists` and remove `isStorageClassExistAfterdelete` (just test for `assert.True(t, errors.IsNotFound(err))`.

**Review by jarrpa — APPROVED** [review]

**Review by nbalacha — COMMENTED** [review]
/lgtm

**Pull request merged**